librad/src/net/peer/storage.rs

```diff
@@ -215,6 +288,15 @@ impl broadcast::LocalStorage<SocketAddr> for Storage {
 
                 Err(e) => match e {
                     Error::KnownObject(_) => PutResult::Stale,
+                    Error::RateLimited { remote_peer, urn } => {
+                        tracing::warn!(
+                            "skipped fetch of {} from {} due to rate limiting",
+                            remote_peer,
+                            urn
+                        );
+                        // Someone else may still be interested
+                        PutResult::Uninteresting
```

Comment:
Isn't a `PutResult::Stale` more meaningful for this branch? Alternatively bubble up the RateLimited as variant of `PutResult`?

Reply:
I think `Stale` means the gossip won't be forwarded, whereas `Uninteresting` will be.

So, correct me if I'm wrong, the semantics of this is that this peer is pummeling me so I'm going to rate-limit them, but some other peer may be able to reply to their gossip (they may not be being pummelled =])

Reply:
Yes that’s right, although if all the pummels hit the limit at the same time, the message will bounce indefinitely. Stale is not correct, either, unless one argues that chatty peers should be punished by not forwarding their logorrhoea. Perhaps a delay would work, but that consumes resources on the wrong end.